### PR TITLE
Improve softmax

### PR DESCRIPTION
**Pull request by rsokl**

### CODE REVIEW

File: src/mygrad/nnet/activations/softmax.py [hunk 1]
```diff
@@ -9,48 +9,48 @@ def _softmax(x, kwargs):
...
     if np.issubdtype(x.dtype, np.integer):
         x = x.astype(np.float)
-    np.exp(x, out=x)
-    x /= x.sum(**kwargs)
+    if x.ndim > 0:
+        np.exp(x, out=x)
+        x /= x.sum(**kwargs)
+    else:
+        x = np.exp(x)
+        x = x / x.sum(**kwargs)
```
Comment (davidmascharka) [inline src/mygrad/nnet/activations/softmax.py:17]: isn't this branch just for scalars? In which case the output of softmax would just be 1 by definition so we don't need to do this computation?

File: src/mygrad/nnet/activations/softmax.py [hunk 2]
```diff
@@ -85,54 +85,50 @@ def softmax(x, constant=False):
...
 
     Parameters
     ----------
-    x : array_like, shape=(D,) or shape=(N,D)
-        log-softmax is computed along the rows of ``x`` if
-        ``x`` is a 2D array. Otherwise log-softmax is computed
-        on the 1D ``x``.
+    x : array_like
+
+    axis : Union[None, int, Tuple[int, ...]], optional (default=-1)
```
Comment (davidmascharka) [inline src/mygrad/nnet/activations/softmax.py:120]: for these can we specify the `axis` as a `mg.Tensor` or `np.ndarray`? do we want to allow that?

File: src/mygrad/nnet/activations/softmax.py (hunk 1 above)
Comment (rsokl) [reply src/mygrad/nnet/activations/softmax.py:17]: oh... ha. You are right

File: src/mygrad/nnet/activations/softmax.py (hunk 2 above)
Comment (rsokl) [reply src/mygrad/nnet/activations/softmax.py:120]: At that point I might as well put `Sequence[int]`. I think I'll just keep it simple though

File: src/mygrad/nnet/activations/softmax.py (hunk 1 above)
Comment (rsokl) [inline src/mygrad/nnet/activations/softmax.py:17]:
```suggestion
        x = np.ones_like(x)
```

**Pull request merged**